File: test/org.eclipse.vjet.test.core.jstojava/testFiles/vjoPro/samples/otype/MShape.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
//<needs(vjoPro.samples.otype.Shape)
vjo.mtype('vjoPro.samples.otype.MShape')
.props({

	line : null, // < Shape::Line

	//>public boolean isOrigin(Shape::Point)
	isOrigin : function(point) {
		return point.x === 0 && point.y === 0;
	},

	//> public Shape.Line createLine(Shape:Point, Shape:Point)
	createLine : function(pt1, pt2) {
		this.line = {
			start : {
				x : pt1.x,
				y : pt1.y
			},
			end : {
				x : pt2.x,
				y : pt2.y
			}
		};
		return this.line;
	}
}).endType();

File: test/org.eclipse.vjet.test.core.jstojava/src/org/eclipse/vjet/dsf/jst/validation/vjo/vjoPro/samples/otype/MShape.java
/*******************************************************************************
 * Copyright (c) 2005-2012 eBay Inc.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 *******************************************************************************/

package org.eclipse.vjet.dsf.jst.validation.vjo.vjoPro.samples.otype;




import java.util.List;

import org.eclipse.vjet.dsf.jsgen.shared.ids.FieldProbIds;
import org.eclipse.vjet.dsf.jsgen.shared.ids.MethodProbIds;
import org.eclipse.vjet.dsf.jsgen.shared.ids.TypeProbIds;
import org.eclipse.vjet.dsf.jsgen.shared.validation.vjo.VjoSemanticProblem;
import org.eclipse.vjet.dsf.jst.validation.vjo.VjoValidationBaseTester;
import org.junit.Before;
import org.junit.Test;




/**
 * MShape.java
 * 
 * @author <a href="mailto:liama@ebay.com">liama</a>
 * @since JDK 1.5
 */
//@Category( { P3, FAST, UNIT })
//@ModuleInfo(value="DsfPrebuild",subModuleId="JsToJava")
public class MShape extends VjoValidationBaseTester {

    @Before
    public void setUp() {
        expectProblems.clear();
        expectProblems
                .add(createNewProblem(TypeProbIds.UnusedActiveNeeds, 1, 0));
        expectProblems.add(createNewProblem(MethodProbIds.ShouldReturnValue,
                15, 0));
        expectProblems
                .add(createNewProblem(FieldProbIds.UndefinedField, 16, 0));
        expectProblems.add(createNewProblem(TypeProbIds.TypeMismatch, 9, 0));
    }

    @Test
    //@Category( { P3, FAST, UNIT })
    //@Description("To test VjoPro project false positive")
    public void testMShape() {
        List<VjoSemanticProblem> problems = getVjoSemanticProblem(
                "vjoPro.samples.otype.", "MShape.js", this.getClass());
        assertProblemEquals(expectProblems, problems);
    }
}
